src/CMakeLists.txt

```diff
@@ -93,57 +227,609 @@ if(MSVC)
 endif()
 
 if("${HPX_PLATFORM_UC}" STREQUAL "ANDROID")
-  add_hpx_library_sources(hpx_external GLOB GLOBS "${ANDROID_NDK_ROOT}/sources/android/cpufeatures/cpu-features.c")
+  set(hpx_external_SOURCES ${hpx_external_objects_SOURCES}
+    "${ANDROID_NDK_ROOT}/sources/android/cpufeatures/cpu-features.c")
 endif()
 
 ################################################################################
 # gather headers
 
-# libhpx headers
-add_hpx_library_headers(hpx
-  GLOB GLOBS "${PROJECT_SOURCE_DIR}/hpx/*.hpp"
-  EXCLUDE ".*hpx_main.*[.]hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/runtime/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/config/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/include/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/lcos/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/performance_counters/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/traits/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/util/*.h*"
-  EXCLUDE "(.*_binary_.*|[io]chunk_.*)[.]hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB GLOBS "${PROJECT_SOURCE_DIR}/hpx/plugins/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/parallel/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/compute/*.hpp"
-  APPEND)
-add_hpx_library_headers(hpx
-  GLOB_RECURSE GLOBS "${PROJECT_SOURCE_DIR}/hpx/compat/*.hpp"
-  APPEND)
-
-# libhpx_init headers
-add_hpx_library_headers(hpx_init
-  GLOB GLOBS "${PROJECT_SOURCE_DIR}/hpx/hpx_main*.hpp")
-
-add_hpx_library_headers(hpx_external GLOB APPEND)
+if(MSVC)
```

Comment:
Why `MSVC` only?

Reply:
Because it's only relevant for `MSVC`. But that's ignored in `add_hpx_source_group` so it doesn't make sense to duplicate that check here. Thanks for noticing! `hpx_SOURCES` below should also be `hpx_HEADERS`.